lib/pure/os.nim

```diff
@@ -1213,11 +1213,11 @@ proc findExe*(exe: string, followSymlinks: bool = true;
         when not defined(windows):
           while followSymlinks: # doubles as if here
             if x.checkSymlink:
-              var r = newString(256)
-              var len = readlink(x, r, 256)
+              var r = newString(1024)
+              var len = readlink(x, r, 1024)
               if len < 0:
                 raiseOSError(osLastError(), exe)
-              if len > 256:
+              if len > 1024:
```

Comment:
Shouldn't this be greater than or equal? 

From the [manual page](https://man7.org/linux/man-pages/man2/readlink.2.html)
> On success, these calls return the number of bytes placed in buf.
> (If the returned value equals bufsiz, then truncation may have occurred.)

Which means if `len == 1024` (or 256, or whatever), then you have to deal with possible truncation.

Reply:
> Which means if `len == 1024` (or 256, or whatever), then you have to deal with possible truncation.

Yes, I already stated that. Please see my other comments on the PR.
If you say `> 1024` or `>= 1024` doesn't matter much. If you write `>=` then we get a truncation to `1025` characters with this current code which is no improvement.

From one of my comments above:
> In any case: The code silently does the wrong thing, i.e., returns truncated paths, if the paths are larger than 256/1024/BIG_NUM. Ideally, some kind of error handling/reporting should happen. I am no Nim developer, though and don't know other details of the code, so I cannot address that issue for you.

Reply:
To further clarify:
The current
```nim
                r = newString(len+1)
                len = readlink(x, r, len)
```
does not make sense if there is a truncation of multiple characters.
This code should be replaced by error handling/reporting.

Reply:
Ah! I see. Thanks for your patience. It was my fault for not reading the rest of the original implementation.

Then a proper implementation would be roughly as follows, I think?
```
var buffer = newString(1024)
while true:
  var res = readlink(x, buffer, len(buffer))
  if res < 0:
    raiseOSError(osLastError()
  elif res == len(buffer):
    setLen(buffer, len(buffer) * 2)
    continue
  else:
    setLen(buffer, res)
    break
```

The loop is because it's possible (though highly unlikely) for the link to change length between calls.